plugin/src/main/java/io/jenkins/plugins/analysis/core/portlets/PullRequestMonitoringPortlet.java

```diff
@@ -0,0 +1,130 @@
+     *              the corresponding {@link ResultAction}.
+     */
+    public PullRequestMonitoringPortlet(final ResultAction action) {
+        this.action = action;
+    }
+
+    @Override
+    public String getTitle() {
+        return action.getLabelProvider().getName();
+    }
+
+    @Override
+    @JavaScriptMethod
+    public String getId() {
+        return action.getResult().getId();
+    }
+
+    @Override
+    public int getPreferredWidth() {
+        return 350;
+    }
+
+    @Override
+    public int getPreferredHeight() {
+        return 350;
+    }
+
+    @Override
+    public String getIconUrl() {
+        return action.getIconFileName();
+    }
+
+    @Override
+    public Optional<String> getDetailViewUrl() {
+        return Optional.of("../" + action.getUrlName());
+    }
+
+    /**
+     * Get the json data for the sunburst diagram.
+     *
+     * @return
+     *          the data as json string.
+     */
+    public String getResultIssuesJsonModel() {
+        JsonObject sunburstData = new JsonObject();
+        sunburstData.addProperty("fixed", action.getResult().getFixedIssues().getSize());
+        sunburstData.addProperty("outstanding", action.getResult().getOutstandingIssues().getSize());
+
+        JsonObject newIssues = new JsonObject();
+        newIssues.addProperty("total", action.getResult().getNewIssues().getSize());
+        newIssues.addProperty("low", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_LOW));
+        newIssues.addProperty("normal", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_NORMAL));
+        newIssues.addProperty("high", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_HIGH));
+        newIssues.addProperty("error", action.getResult().getNewIssues().getSizeOf(Severity.ERROR));
+
+        sunburstData.add("new", newIssues);
+
+        return sunburstData.toString();
+    }
+
+    /**
+     * Check if {@link AnalysisResult} has issues to display in sunburst diagram.
+     *
+     * @return
+     *          true if result has issues, else false.
+     */
+    public boolean hasIssues() {
+        return action.getResult().getFixedIssues().getSize() != 0
+                || action.getResult().getNewIssues().getSize() != 0
+                || action.getResult().getOutstandingIssues().getSize() != 0;
+    }
+
+    /**
+     * The factory for the {@link PullRequestMonitoringPortlet}.
+     */
+    @Extension
+    public static class PortletFactory implements MonitorPortletFactory {
+
+        @Override
+        public Collection<MonitorPortlet> getPortlets(final Run<?, ?> build) {
+            List<ResultAction> action = build.getActions(ResultAction.class);
+
+            return action.stream()
+                    .map(PullRequestMonitoringPortlet::new)
+                    .collect(Collectors.toCollection(ArrayList::new));
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Warnings Next Generation";
```

Comment:
This should be part of `Messages.properties` in `io.jenkins.plugins.analysis.core.portlets`.